subprojects/gradle/build-trace/src/main/kotlin/com/avito/android/build_trace/internal/TaskExt.kt

```diff
@@ -0,0 +1,29 @@
+package com.avito.android.build_trace.internal
+
+import org.gradle.api.Task
+import org.gradle.api.internal.TaskInternal
+import org.gradle.api.tasks.TaskDependency
+
+internal val Task.type: Class<out Task>
+    get() = (this as TaskInternal).taskIdentity.type
+
+internal val Task.predecessors: Set<Task>
+    get() {
+        val dependenciesByInputs = project.gradle.taskGraph.getDependencies(this)
```

Comment:
Important knowledge.
This is what we see in a build scan. Didn't find a better way to get them.